src/ai/susi/mind/SusiSkill.java

```diff
@@ -547,65 +567,60 @@ public static JSONObject getSkillMetadata(String model, String group, String lan
+                        return skillName;
                     }
                 }
             }
-            else {
-                JSONObject newRating=new JSONObject();
-                newRating.put("negative", "0");
-                newRating.put("positive", "0");
-                newRating.put("feedback_count", 0);
-
-                JSONObject newFiveStarRating=new JSONObject();
-                newFiveStarRating.put("one_star", 0);
-                newFiveStarRating.put("two_star", 0);
-                newFiveStarRating.put("three_star", 0);
-                newFiveStarRating.put("four_star", 0);
-                newFiveStarRating.put("five_star", 0);
-                newFiveStarRating.put("avg_star", 0);
-                newFiveStarRating.put("total_star", 0);
-
-                newRating.put("stars", newFiveStarRating);
-
-                skillMetadata.put("skill_rating", newRating);
-            }
         }
+        JSONObject newRating=new JSONObject();
+        newRating.put("negative", "0");
+        newRating.put("positive", "0");
+        newRating.put("feedback_count", 0);
+
+        JSONObject newFiveStarRating=new JSONObject();
+        newFiveStarRating.put("one_star", 0);
+        newFiveStarRating.put("two_star", 0);
+        newFiveStarRating.put("three_star", 0);
+        newFiveStarRating.put("four_star", 0);
+        newFiveStarRating.put("five_star", 0);
+        newFiveStarRating.put("avg_star", 0);
+        newFiveStarRating.put("total_star", 0);
+
+        newRating.put("stars", newFiveStarRating);
+
+        return newRating;
+    }
 
-
-        // file attributes
-        BasicFileAttributes attr = null;
-        Path p = Paths.get(skillpath.getPath());
-        try {
-            attr = Files.readAttributes(p, BasicFileAttributes.class);
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        if(attr!=null){
-            skillMetadata.put("creationTime" , attr.creationTime());
-            skillMetadata.put("lastAccessTime" , attr.lastAccessTime());
-            skillMetadata.put("lastModifiedTime" , attr.lastModifiedTime());
+    public static int getSkillUsage(String model, String group, String language, String skillname) {
+        // usage
+        int weekUsage = 0;
+        JsonTray skillUsage = DAO.skillUsage;
+        if (skillUsage.has(model)) {
+            JSONObject modelName = skillUsage.getJSONObject(model);
+            if (modelName.has(group)) {
+                JSONObject groupName = modelName.getJSONObject(group);
+                if (groupName.has(language)) {
+                    JSONObject languageName = groupName.getJSONObject(language);
+                    if (languageName.has(skillname)) {
+                        JSONArray skillName = languageName.getJSONArray(skillname);
+
+                        // Fetch only last seven days skill usage data for sorting purpose.
+                        int duration = 7;
```

Comment:
Why is this not generic? Please pass the '7' as an argument of the getSkillUsage call